src/factories/request.js

```diff
@@ -73,7 +72,7 @@ const fetchRetry = (
         if (response.ok) {
           resolve(response.json)
         }
-        if (attempt !== maxAttempts && response.status === 429) {
+        if (attempt !== config.retryMaxAttempts && response.status === 429) {
```

Comment:
So it looks like if `attempt` starts with 1 and `retryMaxAttempts` is set to zero this can loop forever. I think you need to initialize attempt with zero and use `<=` here.

Reply:
I changed this to a < 

Reason I did that was because I decided to rename retryMaxAttempts to fetchMaxAttempts.

The reasoning was because it seems easier to count how many times the fetch function runs, rather than, how many times the fetch function has retried. 

Even if someone puts 0. the logic will ensure we always run once.